src/server/game/Entities/Player/Player.cpp

```diff
@@ -6394,7 +6394,7 @@ float getProbabilityOfLevelUp(uint32 SkillValue)
         return 0.0f;
     }
 
-    std::array bounds{ 115, 135, 160, 190, 215, 295, 315, 355, 425, 450 };
+    std::array<uint32, 10> bounds{ 115, 135, 160, 190, 215, 295, 315, 355, 425, 450 };
     std::array<float, 11> dens{ 1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 9.0f, 10.0f, 11.0f, 12.0f, 1.0f };
```

Comment:
Why are you using the `float` type here?
What prevents you from using the `uint32` type

Reply:
```cpp
std::array<uint32, 10> bounds{ 115, 135, 160, 190, 215, 295, 315, 355, 425, 450 };
std::array<uint32, 11> dens{ 1, 2, 3, 4, 5, 6, 9, 10, 11, 12, 1 };
```

Reply:
Well, since I want to return a `float`, I thought that dividing 100 by `float` would lead to a more accurate result. Also, uint32 is such a big data type and completely unnecessary in this scenario. I don't think it makes much sense to use uint32 when I want to return a float.